packages/glib/build.sh

```diff
@@ -67,11 +66,62 @@ termux_step_host_build() {
 termux_step_pre_configure() {
 	# glib checks for __BIONIC__ instead of __ANDROID__:
 	CFLAGS+=" -D__BIONIC__=1"
+	_PREFIX=$TERMUX_PKG_TMPDIR/prefix
+	local _WRAPPER_BIN="${TERMUX_PKG_BUILDDIR}/_wrapper/bin"
+	rm -rf $_PREFIX $_WRAPPER_BIN
+	mkdir -p $_PREFIX $_WRAPPER_BIN
 
-	TERMUX_PKG_VERSION=. termux_setup_gir
+	sed '/^export PKG_CONFIG_LIBDIR=/s|$|:'${_PREFIX}'/lib/pkgconfig|' \
+		"${TERMUX_STANDALONE_TOOLCHAIN}/bin/pkg-config" \
+		> "${_WRAPPER_BIN}/pkg-config"
+	chmod +x "${_WRAPPER_BIN}/pkg-config"
+	export PKG_CONFIG="${_WRAPPER_BIN}/pkg-config"
+	export PATH="${_WRAPPER_BIN}:${PATH}"
+
+	# Magic happens here.
+	# I borrowed nested building method from https://github.com/termux/termux-packages/blob/1244c75380beefc7f7da9744d55aa88df1640acb/x11-packages/qbittorrent/build.sh#L21-L28
+	# and modified termux_step_configure_meson in runtime to make it use another prefix
+	# Also I used advice from here https://github.com/termux/termux-packages/issues/20447#issuecomment-2156066062
+
+	(
+		eval `declare -f termux_step_configure_meson | sed 's|\$TERMUX_PREFIX|'$_PREFIX'|g'`
```

Comment:
What in the actual hell is going on in this line?
- Why do we need the `eval`
- Why are we using backticks?
- Why are we (re?)declaring `termux_step_configure_meson`?

Reply:
This way we are modifying `termux_step_configure_meson` to install stuff to different prefix. This way we are making sure both no-introspection version of `glib` and compile-only version of `gobject-introspection` will not be installed to regular prefix and will be used only during compiling of `glib`.

Reply:
Sorry I didn't see this reply until now.

So `declare -f` isn't re-declaring the function per se.
It prints out the definition and we then do a `sed` substitution on it and and `eval` causes it to be redeclared.

It's possible I'm missing something but couldn't we do this instead?
```bash
TERMUX_PREFIX="$_PREFIX" termux_step_configure_meson
```
I'm not sure why we need to redeclare the function when we can just overwrite the variable.

Reply:
Correct. For some reason I thought it will not work. I was wrong.

Reply:
It's wonderful how much you can over-complicate a problem when you know enough quirks of a language.